pkg/capabilities/v2/chain-capabilities/evm/proto_helpers.go

```diff
@@ -0,0 +1,306 @@
+package evm
+
+import (
+	"errors"
+	"time"
+
+	evmtypes "github.com/smartcontractkit/chainlink-common/pkg/types/chains/evm"
+	valuespb "github.com/smartcontractkit/chainlink-common/pkg/values/pb"
+)
+
+func ConvertAddressesFromProto(addresses [][]byte) []evmtypes.Address {
+	evmAddresses := make([]evmtypes.Address, 0, len(addresses))
+	for _, address := range addresses {
+		if len(address) != 20 {
+			continue // Invalid address length
+		}
+		evmAddresses = append(evmAddresses, evmtypes.Address(address))
+	}
+	return evmAddresses
+}
+
+func convertAddressesToProto(addresses []evmtypes.Address) [][]byte {
+	protoAddresses := make([][]byte, 0, len(addresses))
+	for _, address := range addresses {
+		protoAddresses = append(protoAddresses, address[:])
+	}
+	return protoAddresses
+}
+
+func ConvertHashesFromProto(hashes [][]byte) []evmtypes.Hash {
+	hashesList := make([]evmtypes.Hash, 0, len(hashes))
+	for _, hash := range hashes {
+		if len(hash) != 32 {
+			continue // Invalid hash length
+		}
+		hashesList = append(hashesList, evmtypes.Hash(hash))
+	}
+	return hashesList
+}
+func convertHashesToProto(hashes []evmtypes.Hash) [][]byte {
+	protoHashes := make([][]byte, 0, len(hashes))
+	for _, hash := range hashes {
+		protoHashes = append(protoHashes, hash[:])
+	}
+	return protoHashes
+}
+
+func convertTopicsToProto(topics [][]evmtypes.Hash) []*Topics {
+	protoTopics := make([]*Topics, 0, len(topics))
+	for _, topic := range topics {
+		topicProto := &Topics{Topic: convertHashesToProto(topic)}
+		protoTopics = append(protoTopics, topicProto)
+	}
+	return protoTopics
+}
+
+func ConvertHeadToProto(h evmtypes.Head) *Head {
+	return &Head{
+		Timestamp:   h.Timestamp,
+		BlockNumber: valuespb.NewBigIntFromInt(h.Number),
+		Hash:        h.Hash[:],
+		ParentHash:  h.ParentHash[:],
+	}
+}
+
+var errEmptyHead = errors.New("head is nil")
+
+func ConvertHeadFromProto(head *Head) (evmtypes.Head, error) {
+	if head == nil {
+		return evmtypes.Head{}, errEmptyHead
+	}
+	return evmtypes.Head{
+		Timestamp:  head.GetTimestamp(),
+		Hash:       evmtypes.Hash(head.GetHash()[:]),
```

Comment:
`[:]` slicing is redundant. GetHash already returns bytes. Also, casting to a fixed-size array is not safe and could cause a panic if the sizes mismatch.